types/shelljs/index.d.ts

```diff
@@ -25,537 +30,1173 @@ export function cd(dir?: string): ShellString;
- * @param    options Available options: -n (Suppresses the normal change of directory when removing directories from the stack, so that only the stack is manipulated) -q: Supresses output to the console.
- * @param      dir     Removes the Nth directory (counting from the right of the list printed by dirs), starting with zero.
- * @return         Returns an array of paths in the stack.
- */
-export function popd(options: string, dir: "-N"): ShellArray;
-
-/**
- * When no arguments are given, popd removes the top directory from the stack and performs a cd to the new top directory. The elements are numbered from 0 starting at the first directory listed with dirs; i.e., popd is equivalent to popd +0. Returns an array of paths in the stack.
- * @param    options Available options: -n (Suppresses the normal change of directory when removing directories from the stack, so that only the stack is manipulated) -q: Supresses output to the console.
- * @param    dir     You can only use -N and +N.
- * @return         Returns an array of paths in the stack.
- */
-export function popd(options: string, dir: string): ShellArray;
-
-/**
- * Clears the directory stack by deleting all of the elements.
- * @param      options Clears the directory stack by deleting all of the elements.
- * @return         Returns an array of paths in the stack, or a single path if +N or -N was specified.
- */
-export function dirs(options: "-c"): ShellArray;
+export interface EchoFunction {
+	/**
+	 * Prints string to stdout, and returns string with additional utility methods like .to().
+	 *
+	 * @param options Available options:
+	 *        - `-e`: interpret backslash escapes (default)
+	 *        - `-n`: remove trailing newline from output
+	 * @param text The text to print.
+	 * @return     Returns the string that was passed as argument.
+	 */
+	(options: string, ...text: string[]): ShellString;
+
+	/**
+	 * Prints string to stdout, and returns string with additional utility methods like .to().
+	 *
+	 * @param text The text to print.
+	 * @return     Returns the string that was passed as argument.
+	 */
+	(...text: string[]): ShellString;
+}
 
-/**
- * Display the list of currently remembered directories. Returns an array of paths in the stack, or a single path if +N or -N was specified.
- * @param    options Displays the Nth directory (counting from the left of the list printed by dirs when invoked without options), starting with zero.
- * @return       Returns an array of paths in the stack, or a single path if +N or -N was specified.
- */
-export function dirs(options: "+N"): ShellString;
+export const echo: EchoFunction;
+
+export interface PushDirFunction {
+	/**
+	 * Save the current directory on the top of the directory stack and then cd to dir.
```

Comment:
Up until this point, most (all?) descriptions are in 3rd person (e.g. `Saves` instead of `Save`).
It would be nice to change all descriptions to be consistent (although that is admittedly not directly related to this PR).